core/src/main/java/org/apache/accumulo/core/tabletserver/log/LogEntry.java

```diff
@@ -107,13 +109,18 @@ public int hashCode() {
   }
 
   public static LogEntry fromMetaWalEntry(Entry<Key,Value> entry) {
-    final Value value = entry.getValue();
 
-    String filePath = value.toString();
+    final Text columnQualifier = entry.getKey().getColumnQualifier();
 
-    validateFilePath(filePath);
+    String filePath = columnQualifier.toString();
+
+    String[] parts = filePath.split("/", 2);
```

Comment:
Thank you. Working on that next.

Reply:
@keith-turner `parts[0]` could be something other than `-`, though. The use of `-` was a backwards-compatible choice to substitute a server name that used to be stored in this field. While it's unlikely that it will contain anything else, it is technically possible if somebody upgrades through multiple versions, since there is no upgrade code to get rid of those. Checking for the length being 2 makes sense, but it doesn't really matter, because it will already fail if it's less than 2, and if it's more than 2 (which is expected, actually), they will be combined into the second one, because we specified "at most 2" as a parameter to the split method. So, all you're doing is adding an extra check to fail with one kind of exception instead of another, and making it less robust on upgrade.

Reply:
#4057 should cover improvements to the validation

Reply:
Correction, this isn't covered by #4057, because it was included here. I think maybe there's a small risk of things going wrong if there's an old log entry from before 2.1.0 when the `-` started getting used in place of the field that was previously in that position. I created #4061 as follow up.